### DISCUSSION TITLE
transcripts taking a very long time

### DISCUSSION THREAD

**Discussion by swiecki** [deepgram/community#263] [General help] [answered]
Hi, I'm using the pre-recorded transcription API and I'm making some requests that are taking a really long time, like 5 minutes for a 2 minute audio clip. What's the best way to debug this? Is this because some of the transcription options are expensive (topics/entities/summarize?) or could a mimetype issue cause this? Open to any ideas

**Comment by jjmaldonis** [comment]
Hey @swiecki can you tell us more about your API request? Long response times are not typical unless Whisper is being used, in which case the transcription takes _much_ longer than a model like Nova.

Below is some useful info for us to know:
- What URL are you using? This will include the query parameters, such as `detect_topics=true`.
- What mimetype are you using? and what format is the audio file?
- Do you have a request ID we can look into? Console -> Usage -> Logs
- How are you making the request? i.e. what programming language? are you using one of our SDKs? can you copy/paste your code?

**Comment by swiecki** [comment]
Hi @jjmaldonis thanks so much for your reply and suggestions on how I can start to debug this issue.

here is an example request ID from a request earlier today:
`dbf92de5-9060-4ab4-a7c1-fb884f555c22`

I think its very possible I'm not handling my mimetypes / encodings correctly, could that be causing it?
the mimetype in this case was mp4. I think in some cases I might be passing in "audio/m4a" which is invalid, right?

I'm using the deepgram python library:
``` python
audio_filename = info["path"]
mimetype = "audio/" + info["ext"]
to_return = {}
with open(audio_filename, "rb") as audio_file:
    source = {
        "buffer": audio_file,
        "mimetype": mimetype,
    } 
    print(source)
    response = await dg_client.transcription.prerecorded(
        source,
        {
            "punctuate": True,
            "smart_format": True,
            "profanity_filter": True,
            "detect_topics": True,
            "detect_entities": True,
            "summarize": True,
            "model": "nova",
            "keywords": info["title"].split(" "),
        },
    )
```

**Reply by jjmaldonis** [reply, marked as answer]
Yes, `audio/m4a` is an invalid mimetype, m4a files should use `audio/mp4`. That could cause the slowdown.

Your code looks good.

It may be worth trying different file formats, which will help you narrow down whether the issue is related to the mimetype or file format. Here is an ffmpeg command that will convert file formats: `ffmpeg -i your_audio.m4a -c copy output.wav` - you can change the file extension of `output.wav` to anything you want. Make sure to change the mimetype when making the request with the new file format.

**Reply by swiecki** [reply]
That's super helpful, thank you!
